internal/platform/appkit/transport/http/problems_test.go

```diff
@@ -0,0 +1,166 @@
+	return "conflict"
+}
+
+func (conflictStub) Conflict() bool {
+	return true
+}
+
+func TestDefaultProblemMatchers(t *testing.T) {
+	tests := []struct {
+		err            error
+		expectedStatus int
+	}{
+		{
+			err:            notFoundStub{},
+			expectedStatus: http.StatusNotFound,
+		},
+		{
+			err:            validationStub{},
+			expectedStatus: http.StatusUnprocessableEntity,
+		},
+		{
+			err:            badRequestStub{},
+			expectedStatus: http.StatusBadRequest,
+		},
+		{
+			err:            conflictStub{},
+			expectedStatus: http.StatusConflict,
+		},
+	}
+
+	converter := NewDefaultProblemConverter()
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(fmt.Sprintf("%d", test.expectedStatus), func(t *testing.T) {
+			problem := converter.NewProblem(context.Background(), test.err).(*problems.DefaultProblem)
+
+			if want, have := test.expectedStatus, problem.Status; want != have {
+				t.Errorf("unexpected status\nexpected: %d\nactual:   %d", want, have)
+			}
+
+			if want, have := test.expectedStatus, problem.Code; want != have {
+				t.Errorf("unexpected code\nexpected: %d\nactual:   %d", want, have)
+			}
+
+			if want, have := test.err.Error(), problem.Detail; want != have {
+				t.Errorf("unexpected detail\nexpected: %s\nactual:   %s", want, have)
+			}
+
+			if want, have := test.err.Error(), problem.Message; want != have {
+				t.Errorf("unexpected message\nexpected: %s\nactual:   %s", want, have)
+			}
+
+			if want, have := test.err.Error(), problem.Error; want != have {
+				t.Errorf("unexpected error\nexpected: %s\nactual:   %s", want, have)
+			}
+		})
+	}
+}
+
+type validationWithViolationsStub struct{}
+
+func (validationWithViolationsStub) Error() string {
+	return "validation"
+}
+
+func (validationWithViolationsStub) Validation() bool {
+	return true
+}
+
+func (validationWithViolationsStub) Violations() []string {
+	return []string{
+		"violation",
+	}
+}
+
+func TestDefaultProblemMatchers_ValidationWithViolations(t *testing.T) {
+	converter := NewDefaultProblemConverter()
+
+	err := validationWithViolationsStub{}
+
+	problem := converter.NewProblem(context.Background(), err).(*problems.ValidationProblem)
+
+	if want, have := http.StatusUnprocessableEntity, problem.Status; want != have {
+		t.Errorf("unexpected status\nexpected: %d\nactual:   %d", want, have)
+	}
+
+	if want, have := http.StatusUnprocessableEntity, problem.Code; want != have {
+		t.Errorf("unexpected code\nexpected: %d\nactual:   %d", want, have)
+	}
+
+	if want, have := err.Error(), problem.Detail; want != have {
+		t.Errorf("unexpected detail\nexpected: %s\nactual:   %s", want, have)
+	}
+
+	if want, have := err.Error(), problem.Message; want != have {
+		t.Errorf("unexpected message\nexpected: %s\nactual:   %s", want, have)
+	}
+
+	if want, have := err.Error(), problem.Error; want != have {
+		t.Errorf("unexpected error\nexpected: %s\nactual:   %s", want, have)
+	}
+
+	if want, have := "violation", problem.Violations[0]; want != have {
```

Comment:
```suggestion
	if want, have := err.Violations, problem.Violations; !assert.ElementsMatch(t, want, have) {
```
might better communicate the semantics